controller/src/main/java/org/jboss/as/controller/persistence/ConfigurationFile.java

```diff
@@ -369,7 +370,8 @@ private File determineMainFile(final String rawName, final String name) {
         if (mainName != null) {
             try {
                 final File mainFile = new File(configurationDir, name != null ? name : mainName);
-                if (mainFile.getCanonicalPath().startsWith(configurationDir.getCanonicalPath())) {
+                if (mainFile.getCanonicalPath().startsWith(configurationDir.getCanonicalPath()) ||
+                        Files.isSymbolicLink(Paths.get(mainFile.getPath()))) {
```

Comment:
You also have a toPath() method to convert the File into a Path: https://docs.oracle.com/javase/7/docs/api/java/io/File.html#toPath()